app/scripts/modules/core/pipeline/config/graph/pipeline.graph.component.ts

```diff
@@ -0,0 +1,379 @@
+
+    // track and save the graph scroll position for executions so it doesn't get reset to
+    // zero every second due to repaint.
+    this.$element.on('mousewheel', () => {
+      if (this.execution) {
+        this.pipelineGraphService.xScrollOffset[this.execution.id] = this.$element.scrollLeft();
+      }
+    });
+
+    this.$scope.nodeRadius = 8;
+    this.$scope.rowPadding = 20;
+    this.$scope.graphVerticalPadding = 11;
+    this.$scope.minExecutionGraphHeight = 40;
+    this.$scope.labelOffsetX = this.$scope.nodeRadius + 3;
+    this.$scope.labelOffsetY = this.$scope.nodeRadius + 10;
+    const graphId = this.pipeline ? this.pipeline.id : this.execution.id;
+
+    this.$scope.nodeClicked = (node: IPipelineNode) => {
+      this.onNodeClick(node);
+    };
+
+    this.$scope.highlight = (node: IPipelineNode) => {
+      if (node.isActive) {
+        return;
+      }
+      node.isHighlighted = true;
+      node.parentLinks.forEach((link: IPipelineLink) => {
+        link.isHighlighted = true;
+        this.setLinkClass(link);
+      });
+      node.childLinks.forEach((link: IPipelineLink) => {
+        link.isHighlighted = true;
+        this.setLinkClass(link);
+      });
+    };
+
+    this.$scope.removeHighlight = (node: IPipelineNode) => {
+      if (node.isActive) {
+        return;
+      }
+      node.isHighlighted = false;
+      node.parentLinks.forEach((link) => {
+        link.isHighlighted = false;
+        this.setLinkClass(link);
+      });
+      node.childLinks.forEach((link) => {
+        link.isHighlighted = false;
+        this.setLinkClass(link);
+      });
+    };
+
+    const handleWindowResize = throttle(() => {
+      this.$scope.$evalAsync(this.updateGraph);
+    }, 300);
+
+    const validationSubscription = this.pipelineConfigValidator.subscribe((validations) => {
+      this.pipelineValidations = validations;
+      this.updateGraph();
+    });
+
+    this.updateGraph();
+    if (this.shouldValidate) {
+      this.$scope.$watch('pipeline', debounce(() => this.pipelineConfigValidator.validatePipeline(this.pipeline), 300), true);
+    } else {
+      this.$scope.$watch('pipeline', this.updateGraph, true);
+    }
+    this.$scope.$watch('viewState', this.updateGraph, true);
+    this.$scope.$watch('execution', this.updateGraph, true);
+    this.$(this.$window).bind('resize.pipelineGraph-' + graphId, handleWindowResize);
+
+    this.$scope.$on('$destroy', () => {
+      validationSubscription.unsubscribe();
+      this.$(this.$window).unbind('resize.pipelineGraph-' + graphId);
+    });
+  }
+}
+
+class PipelineGraphComponent implements ng.IComponentOptions {
+  public bindings: any = {
+    pipeline: '=',
+    execution: '=',
+    viewState: '=',
+    onNodeClick: '=',
+    shouldValidate: '=',
+  };
+
+  public templateUrl: string = require('./pipelineGraph.component.html');
+  public controller: any = PipelineGraphController;
+  public controllerAs = '$ctrl';
```

Comment:
This is the default value for `controllerAs` on components.  No need to specify explicitly with the same value.